beacon-chain/blockchain/execution_engine.go

```diff
@@ -119,6 +123,41 @@ func (s *Service) notifyForkchoiceUpdate(ctx context.Context, headState state.Be
 	return payloadID, nil
 }
 
+// getFinalizedHash returns the finalized payload hash given the finalized root.
+// if the finalized payload is before the first finalized block, it returns the zero hash.
+func (s *Service) getFinalizedHash(ctx context.Context, finalizedRoot [32]byte) ([]byte, error) {
```

Comment:
```suggestion
func (s *Service) getPayloadHash(ctx context.Context, finalizedRoot [32]byte) ([]byte, error) {
```
This function is the same as `getJustifiedHash`, no need to repeat it.